sdk/core/Azure.Core/src/Serialization/ModelSerializerFormat.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// A format used to specify how a model should be serialized and deserialized.
+    /// </summary>
+    public readonly partial struct ModelSerializerFormat : IEquatable<ModelSerializerFormat>
```

Comment:
Curious: why an extensible enum instead of a [StandardFormat](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.standardformat?view=net-7.0)?

Reply:
@KrzysztofCwalina had a reason for this.  I can't recall what it was.

Reply:
I basically wanted a string. This allows for extensibility, i.e. I can implement the interface on my models and serialize them to protobuf. Then @tg-msft and @m-nash convinced me that an EE is as flexible as a string but provide some intellisense support for common cases.

Reply:
Sounds good.  Per our offline discussion, there is also the point that StandardFormat makes sense where it is so lightweight to serialize/deserialize a value that parsing the format string takes a significant portion of the processing time.  Given this, I'll update MutableJsonDocument to take a string instead of a StandardFormat as well, for consistency.